tests/sentry/workflow_engine/processors/test_workflow.py

```diff
@@ -343,6 +343,58 @@ def test_enqueues_workflow_any_logic_type(self):
         )
         assert project_ids[0][0] == self.project.id
 
+    def test_skips_enqueuing_any(self):
+        # skips slow conditions if the condition group evaluates to True without evaluating them
+        assert self.workflow.when_condition_group
+        self.workflow.when_condition_group.update(
+            logic_type=DataConditionGroup.Type.ANY_SHORT_CIRCUIT
+        )
+
+        self.create_data_condition(
+            condition_group=self.workflow.when_condition_group,
+            type=Condition.EVENT_FREQUENCY_COUNT,
+            comparison={
+                "interval": "1h",
+                "value": 100,
+            },
+            condition_result=True,
```

Comment:
I'm a little confused by this test - why do we not need to evaluate a slow data condition? Like how would we know if it's been met?

Reply:
the logic type is ALL, and the fast condition isn't met. therefore we don't need to evaluate the slow condition because it's not possible for all the conditions to be met

Reply:
oh wait where is the fast condition 👀

Reply:
oh wait sorry i read the wrong test
there is a default fast condition here that is met https://github.com/getsentry/sentry/blob/77eab354d850c33d74b351b86754a55a46591b16/tests/sentry/workflow_engine/test_base.py#L94-L101

because it is met, and the logic_type is ANY, one of the conditions is met so we don't have to evaluate the slow condition

Reply:
ooh ok gotcha